packages/bundler-plugin-core/src/debug-id.ts

```diff
@@ -0,0 +1,163 @@
+import { stringToUUID } from "./utils";
+
+// TODO: Find a more elaborate process to generate this. (Maybe with type checking and built-in minification)
+const DEBUG_ID_INJECTOR_SNIPPET =
+  ';!function(){try{var e="undefined"!=typeof window?window:"undefined"!=typeof global?global:"undefined"!=typeof self?self:{},n=(new Error).stack;n&&(e._sentryDebugIds=e._sentryDebugIds||{},e._sentryDebugIds[n]="__SENTRY_DEBUG_ID__",e._sentryDebugIdIdentifier="sdbid-__SENTRY_DEBUG_ID__")}catch(e){}}();';
+
+export function injectDebugIdSnippetIntoChunk(code: string) {
+  const debugId = stringToUUID(code); // generate a deterministic debug ID
+  const ms = new MagicString(code);
+
+  const codeToInject = DEBUG_ID_INJECTOR_SNIPPET.replace(/__SENTRY_DEBUG_ID__/g, debugId);
+
+  // We need to be careful not to inject the snippet before any `"use strict";`s.
+  // As an additional complication `"use strict";`s may come after any number of comments.
+  const commentUseStrictRegex =
+    /^(?:\s*|\/\*(.|\r|\n)*?\*\/|\/\/.*?[\n\r])*(?:"use strict";|'use strict';)?/;
+
+  if (code.match(commentUseStrictRegex)?.[0]) {
+    // Add injected code after any comments or "use strict" at the beginning of the bundle.
+    ms.replace(commentUseStrictRegex, (match) => `${match}${codeToInject}`);
+  } else {
+    // ms.replace() doesn't work when there is an empty string match (which happens if
+    // there is neither, a comment, nor a "use strict" at the top of the chunk) so we
+    // need this special case here.
+    ms.prepend(codeToInject);
+  }
+
+  return {
+    code: ms.toString(),
+    map: ms.generateMap(),
+  };
+}
+
+export async function prepareBundleForDebugIdUpload(
+  bundleFilePath: string,
+  uploadFolder: string,
+  uniqueUploadName: string,
+  logger: Logger
+) {
+  let bundleContent;
+  try {
+    bundleContent = await util.promisify(fs.readFile)(bundleFilePath, "utf8");
+  } catch (e) {
+    logger.warn(`Could not read bundle to determine debug ID and source map: ${bundleFilePath}`);
+    return;
+  }
+
+  const debugId = determineDebugIdFromBundleSource(bundleContent);
+  if (debugId === undefined) {
+    logger.warn(`Could not determine debug ID from bundle: ${bundleFilePath}`);
+    return;
+  }
+
+  bundleContent += `\n//# debugId=${debugId}`;
+  const writeSourceFilePromise = util.promisify(fs.writeFile)(
+    path.join(uploadFolder, `${uniqueUploadName}.js`),
+    bundleContent,
+    "utf-8"
+  );
+
+  const writeSourceMapFilePromise = determineSourceMapPathFromBundle(
+    bundleFilePath,
+    bundleContent,
+    logger
+  ).then(async (sourceMapPath): Promise<void> => {
+    if (sourceMapPath) {
+      return await prepareSourceMapForDebugIdUpload(
+        sourceMapPath,
+        path.join(uploadFolder, `${uniqueUploadName}.js.map`),
+        debugId,
+        logger
+      );
+    }
+  });
+
+  await writeSourceFilePromise;
+  await writeSourceMapFilePromise;
+}
+
+/**
+ * Looks for a particular string pattern (`sdbid-[debug ID]`) in the bundle
```

Comment:
Why did we chose `sdbid`?

Reply:
Just wanted to use something short. It can be changed at any time since nothing depends on this. Do you think another pattern fits better?

Reply:
I would rather it be `sentry`, but it's no big deal.

Reply:
> I would rather it be sentry, but it's no big deal.

No! good point! :) befc007f1c4399d3d91a2d7dec1ae3c34bc3959a